### PR TITLE
Popup/floating window result display

### PR DESCRIPTION
**Pull request by mk5med**
This PR resolves the feature request in issue #5. This adds functionality to create and manage nvim floating windows for displaying translation results.

The current idea is to make general wrapper methods containing the nvim api call-sets for the creation, management, and closing of floating windows.

### CODE REVIEW

**Comment by mk5med** [comment]
I added general window creation and closing methods, and I introduced two enums for easy colour customization of the foreground and background of windows.
From my tests, the `pop` method now works as described in feature request #5. Before I submit my work, are there any changes to the implementation you would like?

File: rplugin/python3/translate.py [hunk 1]
```diff
@@ -2,7 +2,49 @@
...
+    BLACK = 0
+    DARKBLUE = 1
+    DARKGREEN = 2
+    DARKCYAN = 3
+    DARKRED = 4
+    DARKMAGENTA = 5
+    BROWN = 6
+    GREY = 7
+    DARKGREY = 8
+
```
Comment (yanganto) [inline rplugin/python3/translate.py:21]: Many thanks for this color Enum.  It is helpful to use the color.  It just a question not a request, Why there is an empty line after these 9 elements? Maybe you want to leave this empty line after 8 elements?

**Comment by yanganto** [comment]
Hi @viktorKorolyuk 

Many thanks for your help. 

This PR is really high quality and does much help than I expected.
I have test on this,  it works very very good.

Following ideals may not include in this PR, it is based on you.  I hope you have fun with this project.
- The color maybe exported as options for users.  
- The dark gray maybe better for default pop window background, because most of terminal use dark background.  

Thanks again

File: rplugin/python3/translate.py (hunk 1 above)
Comment (mk5med) [reply rplugin/python3/translate.py:21]: No problem, I am glad it is helpful! I added the space to indicate the difference in the item ordering between the two enums. Would you like the space moved/removed?

**Comment by mk5med** [comment]
Thank you, it was fun to work on this project! 
I can add the two remaining ideas before I submit. For the first idea, would you be able to provide more information on what you are looking for?

File: rplugin/python3/translate.py (hunk 1 above)
Comment (yanganto) [reply rplugin/python3/translate.py:21]: Got it, please keep it.  Thanks for the details.

File: rplugin/python3/translate.py [hunk 2]
```diff
@@ -20,6 +62,10 @@ def __init__(self, nvim):
         # if you are using different language, this may set as ""
         self.v_line_spacer = self.nvim.vars.get('translate_v_line_spacer', " ")
 
+        # NOTE: The config is auto use 'pop' for you
+        # setup display options, status, pop
+        self.display_option = DisplayType.STATUS if self.nvim.vars.get('translate_display_option') == 'status' else DisplayType.POPUP
```
Comment (yanganto) [inline rplugin/python3/translate.py:67]: The colors maybe set as additional options `translate_bg_color` & `translate_fg_color` here.  Thanks :)

Comment (mk5med) [reply rplugin/python3/translate.py:67]: Sounds good! I added support for `translate_display_colortype`, `translate_fg_color`, and `translate_bg_color` variables. `translate_display_colortype` is used to properly map the string colors to the nvim equivalent. It is expected to be either 8 or 16, and the program defaults to 8 if the variable has not been set.

**Comment by yanganto** [comment]
@viktorKorolyuk Many Thanks.

**Pull request merged**